packages/twenty-front/src/modules/workflow/search-variables/components/SearchVariablesDropdown.tsx

```diff
@@ -59,12 +61,44 @@ const SearchVariablesDropdown = ({
 
   const handleSubItemSelect = (subItem: string) => {
     onVariableSelect(subItem);
+    setSelectedStep(undefined);
+    closeDropdown();
   };
 
   const handleBack = () => {
     setSelectedStep(undefined);
   };
 
+  const getSearchVariablesDropdownComponents = () => {
+    if (isDefined(selectedStep) && isDefined(objectNameSingularToSelect)) {
+      return (
+        <SearchVariablesDropdownObjectItems
+          step={selectedStep}
+          onSelect={handleSubItemSelect}
+          onBack={handleBack}
+        />
+      );
+    }
+
+    if (isDefined(selectedStep)) {
+      return (
+        <SearchVariablesDropdownFieldItems
+          step={selectedStep}
+          onSelect={handleSubItemSelect}
+          onBack={handleBack}
+        />
+      );
+    }
+
+    return (
+      <SearchVariablesDropdownWorkflowStepItems
+        steps={availableVariablesInWorkflowStep}
+        onSelect={handleStepSelect}
+        objectNameSingularToSelect={objectNameSingularToSelect}
+      />
+    );
+  };
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🤓 nitpick:** I like when these pseudo React components are functions prefixed with `render`, like `renderSearchVariablesDropdownComponents`.‏